### PR TITLE
Implements Songs recommender

### PR DESCRIPTION
**Pull request by BaltaJmn**
https://github.com/Naxomiun/SpotiWrap/assets/38072145/0588f68e-a795-45d3-b0fd-5902688c2ce0

### CODE REVIEW

File: core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import com.wachon.spotiwrap.data.extensions.toArtistModel
+import com.wachon.spotiwrap.data.worker.Syncable
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+interface SearchRepository : Syncable {
+    suspend fun searchArtist(query: String): Flow<List<ArtistModel>>
+}
+
+class DefaultSearchRepository(
```
Comment (Naxomiun) [inline core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt]: No need of this. UserRepo and ArtistRepo already exists

File: features/recommender/src/main/kotlin/com/wachon/spotiwrap/features/recommender/presentation/RecommenderViewModel.kt [hunk 2]
```diff
@@ -33,5 +40,45 @@ class RecommenderViewModel(
         }
     }
 
+    fun updateQuery(query: String) {
```
Comment (Naxomiun) [inline features/recommender/src/main/kotlin/com/wachon/spotiwrap/features/recommender/presentation/RecommenderViewModel.kt]: What happened with reactivity?

File: core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/service/SpotifyService.kt [hunk 3]
```diff
@@ -45,6 +46,14 @@ class SpotifyService(
         }.body()
     }
 
+    suspend fun getRecommendations(artists: String, tracks: String, genres: String): RecommendationsApi {
+        return httpClient.get("/v1/recommendations") {
```
Comment (Naxomiun) [inline core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/service/SpotifyService.kt]: Maybe you should emit flow from here not in repo

File: core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt (hunk 1 above)
Comment (Naxomiun) [reply core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt]: +1

File: features/recommender/src/main/kotlin/com/wachon/spotiwrap/features/recommender/presentation/RecommenderViewModel.kt [hunk 4]
```diff
@@ -40,12 +42,34 @@ class RecommenderViewModel(
...
+    fun updateGenres(hasToSave: Boolean, genre: String) {
+        if (hasToSave) {
+            addGenre(genre)
+        } else {
+            removeGenre(genre)
+        }
+    }
+
+    private fun addGenre(genre: String) {
+        _uiState.value = _uiState.value.copy(genresChecked = _uiState.value.genresChecked.toMutableList().apply {
```
Comment (Naxomiun) [inline features/recommender/src/main/kotlin/com/wachon/spotiwrap/features/recommender/presentation/RecommenderViewModel.kt]: _uiState.update { }

File: features/recommender/src/main/kotlin/com/wachon/spotiwrap/features/recommender/presentation/components/GenresContent.kt [hunk 5]
```diff
@@ -76,7 +80,12 @@ fun GenresTitle(
 }
 
 @Composable
-fun GenresRow(genres: List<String>, rowsPerColumn: Int, checkedList: SnapshotStateList<String>) {
+fun GenresRow(
+    rowsPerColumn: Int,
+    genres: List<String>,
```
Comment (Naxomiun) [inline features/recommender/src/main/kotlin/com/wachon/spotiwrap/features/recommender/presentation/components/GenresContent.kt]: Take a look on Immutable/Persistent lists

File: core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt (hunk 1 above)
Comment (BaltaJmn) [reply core/data/src/main/kotlin/com/wachon/spotiwrap/data/repository/SearchRepository.kt]: Done, I have already split the methods between Playlist Repo and Artist Repo

File: core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/service/SpotifyService.kt (hunk 3 above)
Comment (BaltaJmn) [reply core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/service/SpotifyService.kt]: Done

**Review by Naxomiun — APPROVED** [review]

**Pull request merged**